### DISCUSSION TITLE
React app in Joomla

### DISCUSSION THREAD

**Discussion by 1337180** [joomla/joomla-cms#37189] [Q&A]
Is this possible? How much JS (react) can be in my php extension? Can I do for example IRC integration?
Thanks much.
JurajB

**Comment by brianteeman** [comment]
anything is possible

**Comment by dgrammatiko** [comment]
https://github.com/joomla/joomla-cms/tree/4.1-dev/administrator/components/com_media is build using Vue 3.0
In a similar fashion you can do a React/Preact/Svelte/LitHtml/Solid/etc (bear in mind that all these frameworks have the same fundamental principle that UI = fn(state), so they're kinda different than the classic event driven apps). Also you would need tooling for the JS, something like: https://github.com/joomla/joomla-cms/blob/4.1-dev/build/build-modules-js/javascript/build-com_media-js.es6.js

**Comment by 1337180** [comment]
hello coderz 
:) I have a question in my advance in programming I had this idea that I build complete big react app and make it together with Joomla! api.
Will it work ? Or do I have stop dreaming about it because of something? Is this generally good idea?
Thanks.

**Reply by 1337180** [reply]
in final will be there still option to insert some extensions from JED? and what about interaction from my react app to this plugins - can you generally tell everything is possible or how much choices I have with react and Joomla! 4.
Thanks.
